src/App.tsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react'
+import SpotifyWebApi from 'spotify-web-api-js'
+import Login from './components/Login'
+import MainPage, { UserData } from './components/MainPage/MainPage'
+import { getTokenFromUrl } from './helper/spotify'
+
 function App() {
-  return <div>Hello World</div>
+  const fallbackUserData: UserData = {
+    country: 'DE',
+    display_name: 'John Doe',
+    product: 'premium',
+  }
+
+  const [token, setToken] = useState<string | null>(null)
```

Comment:
Instead of the `useEffect` and default value of `null`, you can add a function to load the initial value:
And type is not correct, because `getTokenFromUrl` can return `undefined` if the token is not part of the URL.

```suggestion
  const [token, setToken] = useState<string | undefined>(getTokenFromUrl)
```

Now, `setToken` is never called. Thus, it's safe to remove the state:

```suggestion
  const token = getTokenFromUrl();
```

But wait, this will not work because you unset the `window.location.hash`. Thus, you have to remember the value (like you did with the state):

 ```suggestion
  const token = useMemo(getTokenFromUrl, []);
```
https://reactjs.org/docs/hooks-reference.html#usememo